subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -584,6 +585,21 @@ static void isr_rx_aux_chain(void *param)
 		return;
 	}
 
+	if (!trx_cnt || !crc_ok) {
```

Comment:
Do not generate `EVENT_DONE_EXTRA_TYPE_SYNC` in `isr_rx_adv_sync` when LLL scheduling is used for chain PDU reception, this was an oversight in the previous PR review.

Line 466, do not reset `*trx_cnt = 0U;`  this may cause periodic sync being lost due to drift compensation not done if chain PDUs are not received, and we want drift compensation done at the complete end of the radio event (AUX_SYNC_IND plus any AUX_CHAIN_IND reception in LLL scheduling). We do not want to generate done event in between AUX_SYNC_IND and AUX_CHAIN_IND receptions.

Reply:
> Do not generate `EVENT_DONE_EXTRA_TYPE_SYNC` in `isr_rx_adv_sync` when LLL scheduling is used for chain PDU reception, this was an oversight in the previous PR review.

The event is just prepared there, it will be sent at the end of radio event i.e. LLL scheduled sequence.

> Line 466, do not reset `*trx_cnt = 0U;` this may cause periodic sync being lost due to drift compensation not done if chain PDUs are not received, and we want drift compensation done at the complete end of the radio event (AUX_SYNC_IND plus any AUX_CHAIN_IND reception in LLL scheduling). We do not want to generate done event in between AUX_SYNC_IND and AUX_CHAIN_IND receptions.

If I understand correctly what you mean, this is how it works. Timings are stored in done event on reception of ADV_SYNC_IND, then once all LLL scheduled AUX_CHAIN_IND are received for the current radio event, the done event is sent so drift can be calculated properly. The done event is also sent if AUX_CHAIN_IND is not received properly since `lll_isr_cleanup` is called unconditionally. Then ULL will start scheduling of next aux scan radio event if necessary, but it does not affect sync at all.